src/mapeo-project.js

```diff
@@ -345,6 +345,24 @@ export class MapeoProject {
     await Promise.all([this.#coreManager.ready(), this.#ownershipWriteDone])
   }
 
+  /**
+   */
+  async close() {
+    this.#l.log('closing project %h', this.#projectId)
+    // close hypercores for every namespace
+    // remove listeners
+    for await (let namespace of NAMESPACES) {
+      const namespaceCores = this.#coreManager.getCores(namespace)
+      for (let { core } of namespaceCores) {
+        core.removeAllListeners()
+        await core.close()
+      }
+    }
```

Comment:
this probably can be abstracted to a `close()` method on the `CoreManager`, which will do this and some other things that are set up in that class.

Reply:
yeah, I've realized that `CoreManager` already has a `close()` method. It basically iterates over every `core` and closes it. 
I've added a `close()` method to the `Datastore` which basically `.close()` and `.removeAllListener()` over the `#coreIndexer`